### PR TITLE
SLING-9969: UsersEntryHandler and GroupEntryHandler contain hardcoded…

### PR DESCRIPTION
**Pull request by karlpauls**
… users/groups home path and doesn't include system-rel-path

### CODE REVIEW

**Review by anchela — COMMENTED** [review]
tests..... :)
lets add a few tests.... for example UserEntryHandler/GroupEntryHandler: use the default user/group path as defined with jackrabbit.

**Comment by karlpauls** [comment]
@anchela - PTAL

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/GroupEntryHandlerTest.java [hunk 1]
```diff
@@ -70,4 +70,16 @@ public void parseUser() throws Exception {
         TestUtils.createRepoInitExtension(handler, aclManager, path, getClass().getResourceAsStream(path.substring(1)));
         verify(aclManager, never()).addGroup(any(Group.class));
     }
+
+    @Test
+    public void parseGroupWithConfig() throws Exception {
+        String path = "/jcr_root/system/groups/g/V084LLw1ypl2l9G0e28c/.content.xml";
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/handlers/GroupEntryHandlerTest.java:76]: my initial reaction was: but that's the same as the hardcoded default..... it's not but with just omitting the 'home' segment it looks similar. i would suggest to use something entirely different like e.g. the default value in apache jackrabbit oak to make it obvious.

i would also suggest to add a test that verifies that groups outside of the configured path are not matched by the parser and thus are ignored.

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java [hunk 2]
```diff
@@ -18,10 +18,7 @@
 
 import org.apache.sling.feature.Extension;
 import org.apache.sling.feature.ExtensionType;
-import org.apache.sling.feature.cpconverter.accesscontrol.AclManager;
-import org.apache.sling.feature.cpconverter.accesscontrol.DefaultAclManager;
-import org.apache.sling.feature.cpconverter.accesscontrol.SystemUser;
-import org.apache.sling.feature.cpconverter.accesscontrol.User;
+import org.apache.sling.feature.cpconverter.accesscontrol.*;
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java:21]: * import.....

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java [hunk 3]
```diff
@@ -118,6 +115,18 @@ public void testUser() throws Exception {
         verify(aclManager, never()).addSystemUser(any(SystemUser.class));
     }
 
+    @Test
+    public void parseUserWithConfig() throws Exception {
+        String path = "/jcr_root/system/users/a/author/.content.xml";
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java:120]: see above in the group test.... same applies here

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/GroupEntryHandlerTest.java (hunk 1 above)
Comment (karlpauls) [reply src/test/java/org/apache/sling/feature/cpconverter/handlers/GroupEntryHandlerTest.java]: I think the tests are covering that - I did update the path

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java (hunk 2 above)
Comment (karlpauls) [reply src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java]: fixed

File: src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java (hunk 3 above)
Comment (karlpauls) [reply src/test/java/org/apache/sling/feature/cpconverter/handlers/UsersEntryHandlerTest.java]: see above

**Comment by karlpauls** [comment]
@anchela - PTAL

**Review by anchela — APPROVED** [review]
looks good.

**Pull request merged**